### PR TITLE
impl(pubsub): make the modack a root span

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/14020

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14055)

### CODE REVIEW

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc [hunk 1]
```diff
@@ -83,6 +85,39 @@ TEST(TracingPullLeaseManagerImplTest, AsyncModifyAckDeadlineSuccess) {
                          SpanNamed("test-subscription modack"))));
 }
 
+#if OPENTELEMETRY_VERSION_MAJOR >= 2 || \
+    (OPENTELEMETRY_VERSION_MAJOR == 1 && OPENTELEMETRY_VERSION_MINOR >= 13)
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:89]: I wonder if we should move these ifs into `SpanIsRoot()`. And if otel-cpp < 1.13, the matcher would just return true.

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc [hunk 2]
```diff
@@ -83,6 +85,39 @@ TEST(TracingPullLeaseManagerImplTest, AsyncModifyAckDeadlineSuccess) {
...
+TEST(TracingPullLeaseManagerImplTest, CreateRootSpan) {
+  auto span_catcher = InstallSpanCatcher();
+  auto active_span = MakeSpan("active span");
+  auto active_scope = opentelemetry::trace::Scope(active_span);
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:93]: Why? It is weird to create a scope then end the span on the next line of code.

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc [hunk 3]
```diff
@@ -83,6 +85,39 @@ TEST(TracingPullLeaseManagerImplTest, AsyncModifyAckDeadlineSuccess) {
...
+  active_span->End();
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:94]: I imagine this should be done after the modack call

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc [hunk 4]
```diff
@@ -83,6 +85,39 @@ TEST(TracingPullLeaseManagerImplTest, AsyncModifyAckDeadlineSuccess) {
...
+  auto mock = std::make_shared<MockPullLeaseManagerImpl>();
+  EXPECT_CALL(*mock, AsyncModifyAckDeadline(_, _, _, _, _))
+      .WillOnce(Return(ByMove(make_ready_future(Status{}))));
+  auto manager = MakeTracingPullLeaseManagerImpl(std::move(mock), kTestAckId,
+                                                 kTestSubscription);
+  auto stub = std::make_shared<MockSubscriberStub>();
+  auto impl = std::make_shared<testing_util::MockCompletionQueueImpl>();
+  CompletionQueue cq = CompletionQueue(std::move(impl));
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:102]: Seems like `CompletionQueue cq;` would suffice.

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc [hunk 5]
```diff
@@ -83,6 +85,39 @@ TEST(TracingPullLeaseManagerImplTest, AsyncModifyAckDeadlineSuccess) {
...
+  google::pubsub::v1::ModifyAckDeadlineRequest request;
+  request.set_ack_deadline_seconds(
+      static_cast<std::int32_t>(kLeaseExtension.count()));
+
+  auto status =
+      manager->AsyncModifyAckDeadline(stub, cq, context, options, request);
+  EXPECT_STATUS_OK(status.get());
+
+  auto spans = span_catcher->GetSpans();
+
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:115]: nit: atypical newline

**Review by dbolduc — APPROVED** [review]

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc (hunk 3 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:94]: Yup, fixed.

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc (hunk 4 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:102]: Done

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc (hunk 5 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:115]: Fixed

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:93]: Now it is not ended, leaving as is

File: google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_lease_manager_test.cc:89]: Yeah... we might want to refactor this in the future

**Pull request merged**